flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/codegen/calls/CallGenerator.scala

```diff
@@ -43,11 +43,16 @@ object CallGenerator {
     val nullTerm = newName("isNull")
     val resultTypeTerm = primitiveTypeTermForTypeInfo(returnType)
     val defaultValue = primitiveDefaultValue(returnType)
+    val nullCheckTerms = if(operands.size > 0) {
+      operands.map(_.nullTerm).mkString(" || ")
+    } else {
+      nullCheck + ""
+    }
 
     val resultCode = if (nullCheck) {
```

Comment:
I think, we just need to modify this condition to `if (nullCheck && operands.nonEmpty)`.  Because if no operands, we do not care about whether the operand is null.